src/app/ReviewService/ReviewService.cs

```diff
@@ -15,33 +17,75 @@ public sealed class ReviewService<TReviewSettings> : IReviewService<TReviewSetti
 	private readonly IReviewPrompter _reviewPrompter;
 	private readonly IReviewSettingsSource<TReviewSettings> _reviewSettingsSource;
 	private readonly ReviewConditionCallback<TReviewSettings>[] _reviewConditions;
+	private readonly ILogger _logger;
 
 	public ReviewService(
+		ILogger<ReviewService<TReviewSettings>> logger,
 		IReviewPrompter reviewPrompter,
 		IReviewSettingsSource<TReviewSettings> reviewSettingsSource,
 		ReviewConditionCallback<TReviewSettings>[] reviewConditions
 	)
 	{
+		_logger = logger;
 		_reviewPrompter = reviewPrompter;
 		_reviewSettingsSource = reviewSettingsSource;
 		_reviewConditions = reviewConditions;
 	}
 
 	/// <inheritdoc/>
-	public Task TryRequestReview(CancellationToken ct)
+	public async Task TryRequestReview(CancellationToken ct)
 	{
-		throw new NotImplementedException();
+		_logger.LogDebug("Trying to request a review.");
+
+		if (await GetAreConditionsSatisfied(ct))
+		{
+			await _reviewPrompter.TryPrompt();
+			_logger.LogInformation("Review Requested");
```

Comment:
Always end sentences with a period.

Reply:
Also it should be
`Review requested.`
or
`Requested review.`
with proper casing.

Reply:
And a dot at the end.